storage/s3/s3.go

```diff
@@ -45,13 +44,14 @@ type s3Storage struct {
 
 // New method creates an implementation of Storage with S3 as the backend.
 func New(opts *Options) (storage.Storage, error) {
-	var creds *credentials.Credentials
-	if (len(opts.Access) != 0 && len(opts.Secret) != 0){
-		creds = credentials.NewStaticV4(opts.Access, opts.Secret, "")
-	} else {
-		creds = credentials.NewIAM("")
-	}
-	client, err := minio.NewWithCredentials(opts.Endpoint, creds, opts.UseSSL, "")
+	// var creds *credentials.Credentials
+	// if len(opts.Access) != 0 && len(opts.Secret) != 0 {
+	// 	creds = credentials.NewStaticV4(opts.Access, opts.Secret, "")
+	// } else {
+	// 	creds = credentials.NewIAM("")
+	// }
+	// client, err := minio.NewWithCredentials(opts.Endpoint, creds, opts.UseSSL, opts.Region)
+	client, err := minio.NewWithRegion(opts.Endpoint, opts.Access, opts.Secret, opts.UseSSL, opts.Region)
```

Comment:
This won't get merged like it is, it drops the iam role stuff.

Reply:
https://github.com/minio/minio-go/blob/3007bfa46ebc036644e727bb1e0cf3696293de50/api.go#L166-L171

All you have to do is add the opts.Region to the end.

Reply:
The main.go changes are fine and NewWithCredentials has a parameter for region. Honestly the part commented out just needs to be reverted and the empty string parameter replaced by the region.

Happy to merge it @Hashfyre with the changes in plugin.go

Reply:
The main.go changes are fine and NewWithCredentials has a parameter for region. Honestly the part commented out just needs to be reverted and the empty string parameter replaced by the region.

Happy to merge it @Hashfyre with the changes in plugin.go